### ISSUE TITLE
[core] Jadx deobfuscates valid identifiers

### ISSUE THREAD

**Issue by bagipro** [skylot/jadx#1913]
Hey!

The latest version deobfuscates valid identifiers in apps that are not obfuscated. This behavior wasn't present before. Sample app: https://drive.google.com/file/d/1NEIPWFwukzv1BgJDHpaqKZAznMf25lqW/view?usp=sharing

When you open the `kotlin` package, you will see the following contents:
<img width="489" alt="image" src="https://github.com/skylot/jadx/assets/15173437/dbb3b967-5b36-40e4-8130-fb65ff6dbc06">

However, I believe, those classes shouldn't be deobfuscated

**Comment by bagipro** [comment]
@skylot 
I have the following configuration:
```json
{
  "flattenPackage": false,
  "checkForUpdates": false,
  "fontStr": "",
  "smaliFontStr": "",
  "editorThemePath": "/org/fife/ui/rsyntaxtextarea/themes/default.xml",
  "lafTheme": "FlatLaf Light",
  "langLocale": {
    "locale": "en_US"
  },
  "autoStartJobs": false,
  "excludedPackages": "",
  "autoSaveProject": false,
  "showHeapUsageBar": false,
  "alwaysSelectOpened": false,
  "useAlternativeFileDialog": false,
  "codeAreaLineWrap": true,
  "srhResourceSkipSize": 3,
  "srhResourceFileExt": ".xml|.html|.js|.json|.txt",
  "searchResultsPerPage": 50,
  "useAutoSearch": true,
  "keepCommonDialogOpen": true,
  "smaliAreaShowBytecode": false,
  "lineNumbersMode": "AUTO",
  "mainWindowVerticalSplitterLoc": 300,
  "debuggerStackFrameSplitterLoc": 300,
  "debuggerVarTreeSplitterLoc": 700,
  "adbDialogPath": "",
  "adbDialogHost": "localhost",
  "adbDialogPort": "5037",
  "codeCacheMode": "MEMORY",
  "usageCacheMode": "NONE",
  "jumpOnDoubleClick": true,
  "treeWidth": 601,
  "dockLogViewer": true,
  "settingsVersion": 18,
  "skipResources": false,
  "skipSources": false,
  "exportAsGradleProject": false,
  "threadsCount": 8,
  "decompilationMode": "AUTO",
  "showInconsistentCode": true,
  "useImports": false,
  "debugInfo": true,
  "addDebugLines": false,
  "inlineAnonymousClasses": false,
  "inlineMethods": true,
  "moveInnerClasses": true,
  "allowInlineKotlinLambda": true,
  "extractFinally": true,
  "replaceConsts": false,
  "escapeUnicode": true,
  "respectBytecodeAccessModifiers": false,
  "userRenamesMappingsMode": "READ",
  "deobfuscationOn": true,
  "deobfuscationMinLength": 2,
  "deobfuscationMaxLength": 64,
  "generatedRenamesMappingFileMode": "READ",
  "deobfuscationUseSourceNameAsAlias": true,
  "resourceNameSource": "AUTO",
  "useKotlinMethodsForVarNames": "APPLY",
  "renameFlags": [
    "CASE",
    "VALID",
    "PRINTABLE"
  ],
  "integerFormat": "AUTO",
  "fsCaseSensitive": false,
  "cfgOutput": false,
  "rawCfgOutput": false,
  "fallbackMode": false,
  "useDx": true,
  "commentsLevel": "NONE",
  "pluginOptions": {}
}
```

**Comment by skylot** [comment]
Cause of the issue was an enabled `Use source file name as class name alias` option (disabled by default).
And because some classes have the same 'source file name' attribute, name checks undo rename and apply "deobfuscated" names instead. So, don't use this option if you don't really need to (it is very rare to be useful).

Anyway, I commit a fix with additional checks to prevent rename to same or already exists names, this should resolve the issue :+1:

**Comment by bagipro** [comment]
@skylot 
> it is very rare to be useful

Actually, it’s very often when it’s useful! In 10% of apps I see obfuscated package name, but “restored” class name 

But isn’t it an error that the given configuration worked perfectly a few commits ago?

**Comment by skylot** [comment]
> it’s useful! In 10% of apps

Oh, nice! Maybe we can enable this option by default :smile:

> worked perfectly a few commits ago?

I am pretty sure these "few commits" are a big batch of changes I merged into master after 1.4.7 release. These changes contain a big refactoring of deobfuscator and rename visitors to improve API for scripts and plugins :slightly_smiling_face:
